src/navigator/main-stack.tsx

```diff
@@ -65,7 +67,16 @@ export const MainStackScreen = () => {
   useAppLockTimer();
   useBeaconHandler();
 
-  const initAppCheck = () => void dispatch(checkApp.submit());
+  useFirebaseApp();
+  const appCheck = firebase.appCheck();
+
+  const initAppCheck = async () => {
+    try {
+      await appCheck.activate('ignored', false);
+      const appCheckToken = await appCheck.getToken();
+      dispatch(checkApp.submit(appCheckToken.token));
+    } catch {}
+  };
 
   useTimerEffect(initAppCheck, APP_CHECK_INTERVAL);
 
```

Comment:
move all this logic into `useFirebaseApp` hook  
let MainStack be as clear as possible